invisible_cities/cities/hypathia.py

```diff
@@ -0,0 +1,160 @@
+"""
+-----------------------------------------------------------------------
+                                 Hypathia
+-----------------------------------------------------------------------
+
+From ancient Greek ‘Υπατια: highest, supreme.
+
+This city reads true waveforms from detsim and compute pmaps from them
+without simulating the electronics. This includes:
+    - Rebin 1-ns waveforms to 25-ns waveforms to match those produced
+      by the detector.
+    - Produce a PMT-summed waveform.
+    - Apply a threshold to the PMT-summed waveform.
+    - Find pulses in the PMT-summed waveform.
+    - Match the time window of the PMT pulse with those in the SiPMs.
+    - Build the PMap object.
+"""
+import numpy  as np
+import tables as tb
+
+from .. types.ic_types import minmax
+from .. database       import load_db
+
+from .. reco                  import tbl_functions        as tbl
+from .. reco                  import  peak_functions      as pkf
+from .. core.random_sampling  import NoiseSampler         as SiPMsNoiseSampler
+from .. io  .        pmaps_io import          pmap_writer
+from .. io.        mcinfo_io  import       mc_info_writer
+from .. io  .run_and_event_io import run_and_event_writer
+from .. io  .trigger_io       import       trigger_writer
+from .. io  .event_filter_io  import  event_filter_writer
+
+from .. dataflow            import dataflow as fl
+from .. dataflow.dataflow   import push
+from .. dataflow.dataflow   import pipe
+from .. dataflow.dataflow   import sink
+
+from .  components import city
+from .  components import print_every
+from .  components import zero_suppress_wfs
+from .  components import WfType
+from .  components import wf_from_files
+
+from .  irene      import check_empty_pmap
```

Comment:
Surely components which are used by more than one city, belong in `components` rather than any single city?

Reply:
You are right, I have pushed a new commit moving those functions to `components`.